### PR TITLE
fossilize_replay: Add a --pipeline-hash option

### PR DESCRIPTION
**Pull request by jekstrand**
This option allows the user to deterministically select exactly one
pipeline from the database.  This is very useful when using fossilize to
examine compiler behavior of individual shaders.

### CODE REVIEW

File: cli/fossilize_replay.cpp [hunk 1]
```diff
@@ -231,6 +231,8 @@ struct ThreadedReplayer : StateCreatorInterface
 
 		unsigned shader_cache_size_mb = 256;
 
+		const char *pipeline_hash = nullptr;
```
Comment (HansKristian-Work) [inline cli/fossilize_replay.cpp]: This should be std::string, and not part of the ThreadedReplayer::Options since it does not affect the ThreadedReplayer.

Comment (HansKristian-Work) [reply cli/fossilize_replay.cpp]: Alternatively, the Hash uint64_t can be parsed directly when parsing arguments, no need to pass around a string.

Comment (HansKristian-Work) [reply cli/fossilize_replay.cpp]: Looking at it some more, just keep the pipeline_hash option here, and change the type to Hash directly.

Comment (jekstrand) [reply cli/fossilize_replay.cpp]: > Alternatively, the Hash uint64_t can be parsed directly when parsing arguments, no need to pass around a string.

I thought about that but I couldn't figure out how to properly propagate the error from the parser lambda.  I guess I could call exit()?

Comment (HansKristian-Work) [reply cli/fossilize_replay.cpp]: Log and exit() is fine with me.

File: cli/fossilize_replay.cpp [hunk 2]
```diff
@@ -2994,6 +3027,16 @@ int main(int argc, char *argv[])
 	});
 	cbs.add("--num-threads", [&](CLIParser &parser) { replayer_opts.num_threads = parser.next_uint(); });
 	cbs.add("--loop", [&](CLIParser &parser) { replayer_opts.loop_count = parser.next_uint(); });
+	cbs.add("--pipeline-hash", [&](CLIParser &parser) {
+		const char *hash_str = parser.next_string();
+		char *end;
+		replayer_opts.pipeline_hash = strtoll(hash_str, &end, 16);
```
Comment (HansKristian-Work) [inline cli/fossilize_replay.cpp]: Actually, shouldn't this should be strtoull?

Comment (jekstrand) [reply cli/fossilize_replay.cpp]: Yes, it should.  Good catch!  Fixed in the latest version.

**Pull request merged**